crates/ra_ide_db/src/defs.rs

```diff
@@ -82,38 +82,40 @@ pub enum NameClass {
     Definition(Definition),
     /// `None` in `if let None = Some(82) {}`
     ConstReference(Definition),
+    FieldShorthand {
+        local: Local,
+        field: Definition,
+    },
 }
 
 impl NameClass {
     pub fn into_definition(self) -> Option<Definition> {
         match self {
             NameClass::Definition(it) => Some(it),
             NameClass::ConstReference(_) => None,
+            NameClass::FieldShorthand { local: _, field } => Some(field),
         }
     }
 
     pub fn definition(self) -> Definition {
         match self {
             NameClass::Definition(it) | NameClass::ConstReference(it) => it,
+            NameClass::FieldShorthand { local, field: _ } => Definition::Local(local),
```

Comment:
And here I *think* we actually want to return `field`, as this is a somewhat fuzzy function, which returns "something" which is a definition. Admitedly, `into_definition` and `definition` naming is suboptimal.